### ISSUE TITLE
Build fails with --disable-amr

### ISSUE THREAD

**Issue by bwspenc** [libMesh/libmesh#850]
There are issues with systems/fem_context.C, maybe other files too.

**Comment by jwpeterson** [comment]
This should be fixed as of 7ec01f7.

**Comment by andrsd** [comment]
This seems to be an issue again. Here is what we did to fix it. Feel free to apply or use a better fix.

```
diff --git a/src/base/dof_map.C b/src/base/dof_map.C
index ea89e7e4b..0ec7624ac 100644
--- a/src/base/dof_map.C
+++ b/src/base/dof_map.C
@@ -1855,13 +1855,14 @@ void DofMap::extract_local_vector (const NumericVector<Number> & Ug,
                                    const std::vector<dof_id_type> & dof_indices_in,
                                    DenseVectorBase<Number> & Ue) const
 {
+  const unsigned int n_original_dofs = dof_indices_in.size();
+
 #ifdef LIBMESH_ENABLE_AMR
 
   // Trivial mapping
   libmesh_assert_equal_to (dof_indices_in.size(), Ue.size());
   bool has_constrained_dofs = false;
 
-  const unsigned int n_original_dofs = dof_indices_in.size();
   for (unsigned int il=0; il != n_original_dofs; ++il)
     {
       const dof_id_type ig = dof_indices_in[il];
```
